core/events.js

```diff
@@ -288,7 +288,7 @@ Blockly.Events.filter = function(queueIn, forward) {
           event.name == lastEvent.name) {
         // Merge change events.
         lastEvent.newValue = event.newValue;
-      } else if (event.IS_UI_EVENT && event.element == 'click' &&
+      } else if (event.type == Blockly.Events.CLICK &&
```

Comment:
The property `IS_UI_EVENT` exists on all events (true only for ui events).
Here I only check for `type === Blockly.Events.CLICK` because its redundant to check that its also a ui event.